lib/screens/player_screen.dart

```diff
@@ -173,3 +163,107 @@ class _PlayerScreenContent extends StatelessWidget {
     );
   }
 }
+
+enum PlayerHideable {
+  bigPlayButton(14, 14),
+  queueButton(0, 27),
+  progressSlider(0, 14),
+  twoLineTitle(0, 27),
+  codecInfo(0, 20),
+  loopShuffleButtons(96, 0);
+
+  final double width;
+  final double height;
+
+  const PlayerHideable(this.width, this.height);
+}
+
+class PlayerHideableController {
+  final verticalHideOrder = [
+    PlayerHideable.bigPlayButton,
+    PlayerHideable.queueButton,
+    PlayerHideable.codecInfo,
+    PlayerHideable.twoLineTitle,
+    PlayerHideable.progressSlider
+  ];
+
+  List<PlayerHideable> _visible = [];
+  Size? _target;
+
+  void setSizePortrait(Size size) {
+    _visible = List.from(PlayerHideable.values);
+
+    var maxPadding =
+        FinampSettingsHelper.finampSettings.playerScreenCoverMinimumPadding + 8;
+    var targetHeight =
+        size.height + size.width * ((maxPadding / 100.0) * 2 - 1);
```

Comment:
What's the logic behind this calculation?

Reply:
It's just screen height - (album cover height when using maximum padding).  This is the max possible control height to avoid excessive album cover shrinkage.